sentry-android-fragment/src/main/java/io/sentry/android/fragment/SentryFragmentLifecycleCallbacks.kt

```diff
@@ -9,13 +9,30 @@ import androidx.fragment.app.FragmentManager.FragmentLifecycleCallbacks
 import io.sentry.Breadcrumb
 import io.sentry.HubAdapter
 import io.sentry.IHub
+import io.sentry.ISpan
 import io.sentry.SentryLevel.INFO
+import io.sentry.SpanStatus
 
 @Suppress("TooManyFunctions")
 class SentryFragmentLifecycleCallbacks(
-    private val hub: IHub = HubAdapter.getInstance()
+    private val hub: IHub = HubAdapter.getInstance(),
+    val enableFragmentLifecycleBreadcrumbs: Boolean,
+    val enableAutoFragmentLifecycleTracing: Boolean
 ) : FragmentLifecycleCallbacks() {
 
+    constructor(
+        enableFragmentLifecycleBreadcrumbs: Boolean = true,
+        enableAutoFragmentLifecycleTracing: Boolean = false
+    ) : this(
+        hub = HubAdapter.getInstance(),
+        enableFragmentLifecycleBreadcrumbs = enableFragmentLifecycleBreadcrumbs,
+        enableAutoFragmentLifecycleTracing = enableAutoFragmentLifecycleTracing
+    )
+
+    private val isPerformanceEnabled = hub.options.isTracingEnabled && enableAutoFragmentLifecycleTracing
+
+    private val fragmentsWithOngoingTransactions = mutableMapOf<Fragment, ISpan>()
```

Comment:
`h`: Should we use a `WeakReference` for the fragment here?

Reply:
good catch, we use `WeakHashMap` for activities, I will use the same then